go/teams/chain.go

```diff
@@ -2192,48 +2220,60 @@ func (t *teamSigchainPlayer) updateMembership(stateToUpdate *TeamSigChainState,
 	}
 }
 
-func (t *teamSigchainPlayer) updateInvites(stateToUpdate *TeamSigChainState, additions map[keybase1.TeamRole][]keybase1.TeamInvite, cancelations []keybase1.TeamInviteID) {
+func (t *teamSigchainPlayer) updateInvites(stateToUpdate *TeamSigChainState, additions map[keybase1.TeamRole][]keybase1.TeamInvite, cancelations []keybase1.TeamInviteID, teamSigMeta keybase1.TeamSignatureMetadata) {
 	for _, invites := range additions {
 		for _, invite := range invites {
-			stateToUpdate.informNewInvite(invite)
+			stateToUpdate.informNewInvite(invite, teamSigMeta)
 		}
 	}
 	for _, cancelation := range cancelations {
-		stateToUpdate.informCanceledInvite(cancelation)
+		stateToUpdate.informCanceledInvite(cancelation, teamSigMeta)
 	}
 }
 
-func (t *teamSigchainPlayer) completeInvites(stateToUpdate *TeamSigChainState, completed map[keybase1.TeamInviteID]keybase1.UserVersionPercentForm) error {
+func (t *teamSigchainPlayer) completeInvites(stateToUpdate *TeamSigChainState,
+	completed map[keybase1.TeamInviteID]keybase1.UserVersionPercentForm,
+	teamSigMeta keybase1.TeamSignatureMetadata) error {
 	for id := range completed {
-		invite, ok := stateToUpdate.inner.ActiveInvites[id]
-		if !ok {
+		inviteMD, found := stateToUpdate.FindActiveInviteMDByID(id)
+		if !found {
 			// Invite doesn't exist or we don't know about it because invite
 			// links were stubbed. We could do a similar check here that we do
 			// in teamSigchainPlayer.useInvites, but we haven't been doing it
 			// in the past, so we might have links with issues like that in the
 			// wild.
 			continue
 		}
-		isNewStyle, err := IsNewStyleInvite(invite)
+		isNewStyle, err := IsNewStyleInvite(inviteMD.Invite)
 		if err != nil {
 			return err
 		}
 		if isNewStyle {
 			return fmt.Errorf("`completed_invites` for a new-style invite (id: %q)", id)
 		}
-		stateToUpdate.informCompletedInvite(id)
+		stateToUpdate.informCompletedInvite(id, teamSigMeta)
 	}
 	return nil
 }
 
-func (t *teamSigchainPlayer) obsoleteInvites(stateToUpdate *TeamSigChainState, roleUpdates chainRoleUpdates, sigMeta keybase1.SignatureMetadata) {
-	if len(stateToUpdate.inner.ActiveInvites) == 0 {
+func (t *teamSigchainPlayer) obsoleteActiveInvites(stateToUpdate *TeamSigChainState, roleUpdates chainRoleUpdates, sigMeta keybase1.SignatureMetadata) {
+	if len(stateToUpdate.inner.InviteMetadatas) == 0 {
 		return
 	}
 
+	m := make(map[keybase1.UID]struct{})
 	for _, uvs := range roleUpdates {
 		for _, uv := range uvs {
```

Comment:
I wonder if this should be filtering "None" updates here. I know it hasn't before, but I wonder what is correct here.